module/documents/active-effect.mjs

```diff
@@ -56,11 +56,10 @@ export default class ActiveEffect5e extends ActiveEffect {
    */
   determineSuppression() {
     this.isSuppressed = false;
-    if ( this.disabled || (this.parent.documentName !== "Actor") ) return;
-    const [parentType, parentId, documentType, documentId] = this.origin?.split(".") ?? [];
-    if ( (parentType !== "Actor") || (parentId !== this.parent.id) || (documentType !== "Item") ) return;
-    const item = this.parent.items.get(documentId);
-    if ( !item ) return;
+    if ( this.disabled || (this.parent.documentName !== "Actor") || !this.origin ) return;
+    // Determine if this is an effect from an item
+    const item = fromUuidSync(this.origin);
+    if ( item?.parent !== this.parent ) return;
```

Comment:
Partly my fault for not giving you fully-baked code suggestions, but `fromUuidSync` can throw an error in some cases (e.g., if the UUID points to an embedded item within a document stored in a compendium), so we probably need to wrap this in a `try/catch`.

Reply:
In those cases, is there a preference for logging the error, or should it in this case simply `return`?

Reply:
It should just `return` and treat the item as not suppressed. It's usually a mistake if the `origin` ended up in that state.